zebra/zebra_vrf.c

```diff
@@ -511,3 +511,19 @@ void zebra_vrf_init(void)
 
 	vrf_cmd_init(vrf_config_write, &zserv_privs);
 }
+
+struct zebra_vrf *zvrf_lookup_from_attribute(uint32_t tableid, ns_id_t ns_id)
+{
+	struct vrf *vrf;
+	struct zebra_vrf *zvrf;
+
+	RB_FOREACH (vrf, vrf_name_head, &vrfs_by_name) {
+		zvrf = vrf->info;
+		if (zvrf->table_id != tableid)
+			continue;
+		if (zvrf->zns && zvrf->zns->ns_id != ns_id)
```

Comment:
the logic here made me sort of look twice. I think a NULL `zvrf->zns` will fall through and result in "success", but that's not really what you want? maybe something like this would be more explicit:
```
if (zvrf->zns && zvrf->zns->ns_id == ns_id)
    return zvrf;
else
    continue;
```

Reply:
I think there are 2 ways of writing this.
I tried to align with yours